src/myrtlespeech/run/callbacks/callback.py

```diff
@@ -429,6 +429,10 @@ def on_batch_end(self) -> bool:
         self.state_dict["epoch_batches"] += 1
         if self.training:
             self.state_dict["total_train_batches"] += 1
+
+        # Explicitly delete Tensors from state-dict to save memory
+        self.__clear_memory()
+
```

Comment:
Make this a separate `Callback` rather than forcing it upon the user. There may be a time where it's useful to have access to the `last_input`, `last_target`, `last_output` or `last_loss` in the next batch.

Reply:
Added as a separate Callback